tpm2/encoding_test.go

```diff
@@ -371,3 +371,39 @@ func TestECCParamsEncodeDecode(t *testing.T) {
 		t.Fatalf("got: %+v\nwant: %+v", got, params)
 	}
 }
+
+func TestSignEncode(t *testing.T) {
+	t.Run("AlgNull", func(t *testing.T) {
+		testCmdBytes, err := hex.DecodeString("800000010000000d40000009000001000401020304000301020300108024400000070000")
+		if err != nil {
+			t.Fatal(err)
+		}
+		digest := []byte{0x01, 0x02, 0x03}
+		cmdBytes, err := encodeSign(tpmutil.Handle(0x80000001), defaultPassword, digest, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !bytes.Equal(cmdBytes, testCmdBytes) {
+			t.Fatalf("got: %s, want: %s", hex.EncodeToString(cmdBytes), hex.EncodeToString(testCmdBytes))
+		}
+	})
+
+	t.Run("AlgSHA256", func(t *testing.T) {
+		// TODO: this test string generated by own test. please see https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-3-Commands-01.38.pdf 20.2.2 Command and Respons.
```

Comment:
Not sure that we need "TODO", there's nothing actionable here.
How about: `// The expected output blobs were recorded while successfully running Sign against a real TPM device.`